[IOSP-389] Setting up RubyMine (#293)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/RubyMine.md

# RubyMine

If you have to work with Ruby, i.e. to update our fastlane setup, you better use an IDE that comes with some quality of life tools other than syntax higlighting.
We are extensively using JetBrains products in the company so our license includes such Ruby IDE as RubyMine.

> To install it go to https://www.jetbrains.com/ruby/download/#section=mac to download the macOS app.

Reviewer: ```suggestion
To install it, go to the [JetBrains website](https://www.jetbrains.com/ruby/download/#section=mac) to download the macOS app.
```

> When asked about license enter the address of our license service. The service address is in 1Password. This service is only available from office network or via VPN.

Reviewer: ```suggestion
When asked about license enter the address of our license service. The service address is in 1Password. This service is only available from the office network or via VPN.
```

![](Assets/RubyMineLicense.png)

> Then open the root folder of our project. As most of the files are not in Ruby IDE will spend a lot of time tring to index files we don't need.

Reviewer: ```suggestion
Then open the root folder of our project. As most of the files are not in Ruby, the IDE will spend a lot of time trying to index files we don't need.
```

> To avoid that go to **Preferences -> Project Structure** and select all the foldres except `fastlane` folder as **excluded**.

Reviewer: ```suggestion
To avoid that, go to **Preferences -> Project Structure** and select all the folders except `fastlane` as **excluded**.
```

Reviewer 2: Quick question, wouldn't it be easier to just open the `fastlane/` folder in the IDE, instead of opening the project's root folder and excluding everything but `fastlane/`?

Author: it won't recognise installed gems because Gemfile is not in the current directory

![](Assets/RubyMinePreferences.png)

This will allow RubyMine to discover Ruby Gems that we have defined in the Gemfile in the root directory and only index fastlane files.

> Now RubyMine will give you a hand with code complition for standard Ruby functions, will understand imports of gems with `require`, will suggest you fixes on syntax errors 

Reviewer: ```suggestion
Now RubyMine will give you a hand with code compilation for standard Ruby functions, will understand imports of gems with `require`, will suggest you fixes on syntax errors 
```

Author: Actually I meant `completion` =)
```suggestion
Now RubyMine will give you a hand with code completion for standard Ruby functions, will understand imports of gems with `require`, will suggest you fixes on syntax errors 
```

> Now RubyMine will give you a hand with code completion for standard Ruby functions, will understand imports of gems with `require`, will suggest you fixes on syntax errors 

Reviewer: ```suggestion
Now RubyMine will give you a hand with code completion for standard Ruby functions. It will understand imports of gems with `require`, suggest you fixes for syntax errors 
```
(there's also an unnecessary line break here)

> and other common errors like missing parameters and lint your code style (if you enable scanning with RuboCop).

Reviewer: ```suggestion
, and other common errors like missing parameters. If you enable scanning with RuboCop, it can also lint your code style.
```
